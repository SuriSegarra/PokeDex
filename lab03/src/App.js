import React, { Component } from 'react';
import './App.css';
import Header from './Header.js';
import PokeList from './pokeList';
import request from 'superagent';

export default class App extends Component {
  state = {
    data: []
  }
  async componentDidMount() {
    const data = await request.get('https://alchemy-pokedex.herokuapp.com/api/pokedex')
    
    this.setState({data:data.body.results})
  }
  render(){
    return (
      <div>
        <Header/>
        <PokeList pokemons= {this.state.data}/>
        
      
        
      </div>
    );
  }
}