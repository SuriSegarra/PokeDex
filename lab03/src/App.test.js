import React from 'react';
import renderer from 'react-test-renderer';
import App from './App.js';

test('renders App.js correctly', () => {
  const app = renderer
    .create(<App />)
    .toJSON();
  expect(app).toMatchSnapshot();
});
