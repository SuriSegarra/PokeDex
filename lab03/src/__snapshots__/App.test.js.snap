// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App.js correctly 1`] = `
<div>
  <div>
    <section>
      <h1>
        <a
          href="https://fontmeme.com/pokemon-font/"
        >
          <img
            alt="pokemon-font"
            border="0"
            src="https://fontmeme.com/permalink/200213/54094a75ad0cab6a2805fe58ea555d47.png"
          />
        </a>
      </h1>
    </section>
  </div>
  <div>
    <main>
      <section
        className="options"
      >
        <select
          className="pokemon-filter"
          onChange={[Function]}
        >
          <option
            value="defaultValue"
          >
            What's your POkemon?
          </option>
          <option
            value="grass"
          >
            Grass
          </option>
          <option
            value="fire"
          >
            Fire
          </option>
          <option
            value="water"
          >
            Water
          </option>
        </select>
      </section>
      <section
        className="list-section"
      >
        <ul
          className="pokemons"
        />
      </section>
    </main>
  </div>
</div>
`;
